bin/wasm-node/javascript/src/index.js

```diff
@@ -79,18 +119,54 @@ export async function start(config) {
       // `expected` was pushed by the `addChain` method.
       // Resolve the promise that `addChain` returned to the user.
       expected.resolve({
+        health: () => {
+          // craft a new system_health messsage
+          const id = 'health:' + nextHealthRequestId;
+          nextHealthRequestId += 1;
+          const request = JSON.stringify({ id, jsonrpc: "2.0", method: "system_health", params: [] });
+          let hrResolve, hrReject;
+          const hrPromise = new Promise((resolve, reject) => {
+            hrResolve = resolve;
+            hrReject = reject;
+          });
+
+          // track the new message
+          healthRequestCallbacks.push({
```

Comment:
What I was thinking of is:

```suggestion
healthRequestCallbacks[id] = {
    chainId,
    resolve: hrResolve,
    reject: hrReject
});
```

Reply:
Changing to this shape would make the cleanup code in `remove()` a lot more complicated.

Reply:
How about using a proper [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of an object since that's really what we're doing?

Reply:
Yeah that works